examples/unconditional_image_generation/train_unconditional.py

```diff
@@ -342,19 +342,35 @@ def transforms(examples):
         num_training_steps=(len(train_dataloader) * args.num_epochs),
     )
 
+    if args.use_ema:
+        ema_model = EMAModel(
+            model.parameters(),
+            decay=args.ema_max_decay,
+            use_ema_warmup=True,
+            inv_gamma=args.ema_inv_gamma,
+            power=args.ema_power,
+        )
+
     model, optimizer, train_dataloader, lr_scheduler = accelerator.prepare(
         model, optimizer, train_dataloader, lr_scheduler
     )
-    accelerator.register_for_checkpointing(lr_scheduler)
+    if args.use_ema:
+        accelerator.register_for_checkpointing(ema_model, lr_scheduler)
```

Comment:
```suggestion
        accelerator.register_for_checkpointing(ema_model)
```

Not required as verified by @patil-suraj in https://github.com/huggingface/diffusers/pull/1834#discussion_r1059397718